azure-iot-device/azure/iot/device/iothub/pipeline/iothub_pipeline.py

```diff
@@ -101,12 +111,12 @@ def _on_disconnected():
 
         if isinstance(auth_provider, X509AuthenticationProvider):
             op = pipeline_ops_iothub.SetX509AuthProviderOperation(
-                auth_provider=auth_provider, callback=callback
-            )
+                auth_provider=auth_provider, websockets=self._config.websockets, 
+                callback=callback)
```

Comment:
>a [](start = 26, length = 1)

You have the pipeline extracting the "websockets" value at the top and then passing it down.  This is good, but it means you have to update the pipeline internals to pass it from the top to the bottom.  Think about adding more pipeline options in the future -- do you want to add more lines  to pipeline_stages_iothub for every single option you add.   Instead, I might suggest adding the entire structure as a member of PipelineRootStage.  This way, all stages can get to it via self.pipeline_root and you can acces the attribute only from the stage that actually cares about the value.

Reply:
this has been done now by adding a configurations object to the PipelineRootStage initialization as a parameter.